streams/src/main/java/org/apache/kafka/streams/state/internals/Segment.java

```diff
@@ -25,4 +26,35 @@
 
     void destroy() throws IOException;
 
+    /**
+     * INTERNAL USE ONLY - Move this method to ReadOnlyKeyValueStore to make it a public API
+     *
+     * Get an iterator over a given range of keys. This iterator must be closed after use.
+     * The returned iterator must be safe from {@link java.util.ConcurrentModificationException}s
+     * and must not return null values.
+     * Order is not guaranteed as bytes lexicographical ordering might not represent key order.
+     *
+     * @param from The first key that could be in the range, where iteration starts from.
+     * @param to   The last key that could be in the range, where iteration ends.
+     * @param prefixScan If true, then it iterates using the common key prefixes.
+     * @return The iterator for this range, from smallest to largest bytes.
+     * @throws NullPointerException       If null is used for from or to.*
+     */
+    KeyValueIterator<Bytes, byte[]> range(Bytes from, Bytes to, boolean prefixScan);
+
+    /**
+     * INTERNAL USE ONLY - - Move this method to ReadOnlyKeyValueStore to make it a public API
+     *
+     * Get a reverse iterator over a given range of keys. This iterator must be closed after use.
+     * The returned iterator must be safe from {@link java.util.ConcurrentModificationException}s
+     * and must not return null values.
+     * Order is not guaranteed as bytes lexicographical ordering might not represent key order.
+     *
+     * @param from The first key that could be in the range, where iteration ends.
+     * @param to   The last key that could be in the range, where iteration starts from.
+     * @param prefixScan If true, then it iterates using the common key prefixes.
+     * @return The reverse iterator for this range, from largest to smallest key bytes.
+     * @throws NullPointerException       If null is used for from or to.
+     */
+    KeyValueIterator<Bytes, byte[]> reverseRange(Bytes from, Bytes to, boolean prefixScan);
```

Comment:
Related to the other comment: since in stream-stream join we do not really need reverse-prefixScan, just adding a forward `prefixScan(..)` interface may be better.